main.tf

```diff
@@ -1,24 +1,111 @@
-terraform {
-  required_providers {
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "public1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.5.0/24"
+
+  availability_zone = "us-east-1a"
 
-    aws = {
-      source  = "hashicorp/aws"
-      version = "~> 4.16"
-    }
+  tags = {
+    Name = "10.0.5.0/24_us-east-1a"
   }
-  required_version = ">= 1.2.0"
 }
 
-provider "aws" {
-  region  = "us-east-1"
-  profile = "emanuel"
+resource "aws_subnet" "public2" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.2.0/24"
+
+  availability_zone = "us-east-1b"
+
+  tags = {
+    Name = "10.0.2.0/24_us-east-1b"
+  }
 }
 
-resource "aws_instance" "app_server" {
-  ami           = "ami-02f3f602d23f1659d"
-  instance_type = "t2.micro"
+resource "aws_subnet" "private1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.3.0/24"
+
+  availability_zone = "us-east-1a"  
 
   tags = {
-    Name = "first_instance_as_code"
+    Name = "10.0.3.0/24_us-east-1a"
   }
 }
+
+resource "aws_subnet" "private2" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.4.0/24"
+
+  availability_zone = "us-east-1b"  
+
+  tags = {
+    Name = "10.0.4.0/24_us-east-1b"
+  }
+}
+
+resource "aws_internet_gateway" "main_ig" {
+  vpc_id = aws_vpc.main.id
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.main.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.main_ig.id
+  }
+}
+
+resource "aws_route_table_association" "public1" {
+  subnet_id      = aws_subnet.public1.id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "public2" {
+  subnet_id      = aws_subnet.public2.id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_eip" "nat1" {
+  vpc      = true
+}
+
+resource "aws_eip" "nat2" {
+  vpc      = true
+}
+
+resource "aws_nat_gateway" "nat1" {
+  allocation_id = aws_eip.nat1.id
+  subnet_id     = aws_subnet.public1.id
+}
+
+resource "aws_nat_gateway" "nat2" {
+  allocation_id = aws_eip.nat2.id
+  subnet_id     = aws_subnet.public2.id
+}
+
+resource "aws_route_table" "private" {
```

Comment:
you are to associate the resource "aws_route_table" "private" with the nar gateway and they should be 2 for the two nat gateway each.